docs/operating/concepts/podopslifecycle.md

```diff
@@ -0,0 +1,51 @@
+---
+sidebar_position: 2
+---
+# PodOpsLifecycle
+
+Kubernetes provides a set of default controllers for workload management, like StatefulSet, Deployment, DaemonSet for instances. While user services around pods have difficulty to participate in the operation lifecycle of a pod.
+
+PodOpsLifecycle attempts to provide Kubernetes administrators and developers with finer-grained control the entire lifecycle of a pod. For example, we can develop a controller to do some necessary things in both the pre traffic off and pre traffic on phases to avoid traffic loss.
+
+## Goals
+
+1. Provides extensibility that allows users to control the whole lifecycle of pods using the PodOpsLifecycle mechanism.
+2. Provide some concurrency, multi controllers can operate the pod in the same time. For example, when a pod is going to be replaced, other controllers may want to delete it.
+3. All the lifecycle phases of a pod can be tracing.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As a developer that focus on pod traffic, I should remove the endpoint once the label `prepare.podopslifecycle.kusionstack.io/<id>=<time>` added which means traffic to the pod should be turned off, and I should add the endpoint once the label `complete.podopslifecycle.kusionstack.io/<id>=<time>` added which means traffic to the pod should be turned on.
```

Comment:
1. `focus` => `focuses`

2. traffic controller developer need not to be aware of the pod ops lifecycle labels